src/javascript/typescript/language.ts

```diff
@@ -69,109 +68,108 @@ export default class TypescriptGenerator {
-      })
-    );
+    return fields.map(({name, description, type}) => {
+      const propertySignatureType = t.TSPropertySignature(
+        t.identifier(
+          // Nullable fields on input objects do not have to be defined
+          // as well, so allow these fields to be "undefined"
+          (keyInheritsNullability && this.isNullableType(type))
+            ? name + '?'
+            : name
+        ),
+        t.TSTypeAnnotation(type)
+      );
 
-    return objectTypeAnnotation;
+      if (description) {
+        propertySignatureType.trailingComments = [{
+          type: 'CommentLine',
+          value: ` ${description.replace('\n', ' ')}`
+        } as t.CommentLine]
+      }
+
+      return propertySignatureType;
+    });
   }
 
-  public typeAliasObject(name: string, fields: ObjectProperty[], {
+  public interface(name: string, fields: ObjectProperty[], {
     keyInheritsNullability = false
   }: {
     keyInheritsNullability?: boolean
   } = {}) {
-    return t.typeAlias(
+
+    return t.TSInterfaceDeclaration(
       t.identifier(name),
       undefined,
-      this.objectTypeAnnotation(fields, {
-        keyInheritsNullability
-      })
+      undefined,
+      t.TSInterfaceBody(
+        this.typesForProperties(fields, {
+          keyInheritsNullability
+        })
+      )
     );
   }
 
   public typeAliasObjectUnion(name: string, members: ObjectProperty[][]) {
-    return t.typeAlias(
+    return t.TSTypeAliasDeclaration(
       t.identifier(name),
       undefined,
-      t.unionTypeAnnotation(
+      t.TSUnionType(
         members.map(member => {
-          return this.objectTypeAnnotation(member)
+          return t.TSTypeLiteral(this.typesForProperties(member))
         })
       )
     )
   }
 
-  public typeAliasGenericUnion(name: string, members: t.FlowTypeAnnotation[]) {
-    return t.typeAlias(
+  public typeAliasGenericUnion(name: string, members: t.TSType[]) {
+    return t.TSTypeAliasDeclaration(
       t.identifier(name),
       undefined,
-      t.unionTypeAnnotation(members)
+      t.TSUnionType(
+        members
+      )
     );
   }
 
   public exportDeclaration(declaration: t.Declaration) {
     return t.exportNamedDeclaration(declaration, []);
   }
 
-  public annotationFromScopeStack(scope: string[]) {
-    return t.genericTypeAnnotation(
-      t.identifier(
-        scope.join('_')
-      )
-    );
+  public nameFromScopeStack(scope: string[]) {
+    return scope.join('_');
   }
 
-  public makeNullableAnnotation(annotation: t.FlowTypeAnnotation) {
-    return t.unionTypeAnnotation([
-      annotation,
-      // @ts-ignore
-      t.tSUndefinedKeyword(),
-      // @ts-ignore
-      t.tSNullKeyword()
+  public makeNullableType(type: t.TSType) {
+    return t.TSUnionType([
+      type,
+      // t.TSUndefinedKeyword(),
+      t.TSNullKeyword()
     ])
   }
+
+  public isNullableType(type: t.TSType) {
+    return t.isTSUnionType(type) && t.isTSNullKeyword(type.types[1]);
```

Comment:
oh cool -- would we want this to check `type.types.some(t.isTSNullKeyword)` just in case?

Reply:
Yeah, probably more robust, just didn't do it since we are generating the types ourselves, and the `makeNullableType` method already puts the `null` in the end. But will do it as you suggest